zcash_primitives/CHANGELOG.md

```diff
@@ -51,6 +51,24 @@ and this library adheres to Rust's notion of
     data, as described above.
 - `zcash_primitives::serialize` has been factored out as a new `zcash_encoding`
   crate, which can be found in the `components` directory.
+- `zcash_primitives::zip32::diversifier` has been renamed to `find_sapling_diversifier`
+  and `sapling_diversifier` has been added. `find_sapling_diversifier` searches the
+  diversifier index space, whereas `sapling_diversifier` just attempts to use the
+  provided diversifier index and returns `None` if it does not produce a valid
+  diversifier.
+- `zcash_primitives::zip32::DiversifierKey::diversifier` has been renamed to 
+  `find_diversifier` and the `diversifier` method has new semantics.
+  `find_diversifier` searches the diversifier index space to find a diversifier
+  index which produces a valid diversifier, whereas `diversifier` just attempts
+  to use the provided diversifier index and returns `None` if it does not
+  produce a valid diversifier.
+- `zcash_primitives::zip32::ExtendedFullViewingKey::address` has been renamed
```

Comment:
These types should really be moved to the `sapling` module (which should itself be extracted to a separate crate, similar to how we've structured the relationship with Orchard). This will then mitigate concerns that might be introduced by this renaming/semantic change.